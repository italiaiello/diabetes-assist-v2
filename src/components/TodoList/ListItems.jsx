import React from 'react';
import { ref, remove } from "firebase/database";
import TaskTick from '../../images/Tick.svg'
import { auth, database } from '../../firebase';

const ListItems = props => {

    const removeItem = taskToDelete => {
        let newTasks = props.todoTasks.slice()
        newTasks = newTasks.filter(task => task.id !== taskToDelete.id)
        remove(ref(database, `users/${auth.currentUser.uid}/tasks/${taskToDelete.id}`))
        props.setTodoTasks(newTasks)
    }

    const listItems = props.todoTasks === null ?
        <p id="noTasksMessage">Nothing to do yet</p>
        :
        props.todoTasks.map((taskTodo) => {
            return (
                <div className="list" key={taskTodo.id}>
                    <div className="circleButton" onClick={() => removeItem(taskTodo)}>
                        <figure className="tickIcon">
                            <img src={TaskTick} alt="Task ticked off" />
                        </figure>
                    </div>
                    <p>{taskTodo.task}</p>
                </div>
            )
        })
    

    return (
        <div id="thingsToDo">{listItems}</div>
    )
}

export default ListItems;