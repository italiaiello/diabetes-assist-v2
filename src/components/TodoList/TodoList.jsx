import React, { useState } from 'react';
import ListItems from './ListItems'
import { auth, database } from '../../firebase';
import { ref, set, child, push } from "firebase/database";

const TodoList = props => {

    const [currentItem, setCurrentItem] = useState('')

    const handleInput = e => {
        setCurrentItem(e.target.value)
    }

    const addItem = () => {
        const newTaskKey = push(child(ref(database), `users/${auth.currentUser.uid}/tasks`)).key;
        if (currentItem !== "") {
            if (props.todoTasks !== null) {
                const newItems = props.todoTasks.slice()
                newItems.push({ id: newTaskKey, detail: currentItem })
                props.setTodoTasks(newItems)
                
            } else {
                const newItem = [currentItem]
                props.setTodoTasks(newItem)
            }

            addTaskToDatabase(newTaskKey)
        }

        setCurrentItem('')
    }

    const addWithEnterKey = (e) => {
        let code = (e.keyCode ? e.keyCode : e.which);
        if(code === 13) { // Keycode for Enter button
            addItem()
        }
    }

    const addTaskToDatabase = (taskId) => {
        set(ref(database, `users/${auth.currentUser.uid}/tasks/${taskId}`), {
            id: taskId,
            task: currentItem
        })
    }

    return (
        <div>
            <ListItems todoTasks={props.todoTasks} setTodoTasks={props.setTodoTasks}/>
            <article id="addTask">
                <button type="submit" onClick={addItem}>+</button>
                <input type="text" onKeyPress={addWithEnterKey} placeholder="Add a task" 
                    value={currentItem} onChange={handleInput}
                />
            </article>
        </div>
    )
}

export default TodoList;