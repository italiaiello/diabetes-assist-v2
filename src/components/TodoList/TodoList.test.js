import { shallow } from 'enzyme';
import React  from 'react';
import TodoList from './TodoList';

let route = ""

describe('TodoList.js', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TodoList />)
  })

  it('expects to render TodoList component', () => {
    expect(wrapper).toMatchSnapshot()
  })

}) 