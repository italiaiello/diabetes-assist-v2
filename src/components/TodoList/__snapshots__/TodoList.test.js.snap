// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList.js expects to render TodoList component 1`] = `ShallowWrapper {}`;
